examples/fastspeech2/generate.py

```diff
@@ -0,0 +1,140 @@
+
+import mindaudio
+from text import text_to_sequence
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="FastSpeech2 training")
+    parser.add_argument(
+        "--device_target", type=str, default="GPU", choices=("GPU", "CPU", "Ascend")
+    )
+    parser.add_argument("--device_id", "-i", type=int, default=0)
+    parser.add_argument(
+        "--context_mode", type=str, default="py", choices=["py", "graph"]
+    )
+    parser.add_argument(
+        "--config",
+        "-c",
+        type=str,
+        default="recipes/LJSpeech/tts/fastspeech2/fastspeech2.yaml",
+    )
+    parser.add_argument("--text", "-t", type=str, default="this is a test text")
+    parser.add_argument("--restore", "-r", type=str, default="")
+    parser.add_argument("--data_url", default="")
+    parser.add_argument("--train_url", default="")
+    args = parser.parse_args()
+    return args
+
+
+def read_lexicon(lex_path):
+    lexicon = {}
+    with open(lex_path) as f:
+        for line in f:
+            temp = re.split(r"\s+", line.strip("\n"))
+            word = temp[0]
+            phones = temp[1:]
+            if word.lower() not in lexicon:
+                lexicon[word.lower()] = phones
+    return lexicon
+
+
+def preprocess_english(text, hps):
+    text = text.rstrip(punctuation)
+    lexicon = read_lexicon(hps.lexicon_path)
+
+    g2p = G2p()
+    phones = []
+    words = re.split(r"([,;.\-\?\!\s+])", text)
+    for w in words:
+        if w.lower() in lexicon:
+            phones += lexicon[w.lower()]
+        else:
+            phones += list(filter(lambda p: p != " ", g2p(w)))
+    phones = "{" + "}{".join(phones) + "}"
+    phones = re.sub(r"\{[^\w\s]?\}", "{sp}", phones)
+    phones = phones.replace("}{", " ")
+
+    print("Raw Text Sequence: {}".format(text))
+    print("Phoneme Sequence:", (phones))
+    sequence = np.array(text_to_sequence(phones, ["english_cleaners"]))
+
+    return np.array(sequence)
+
+
+def preprocess_mandarin(text, hps):
+    lexicon = read_lexicon(hps.lexicon_path)
+
+    phones = []
+    pinyins = [
+        p[0]
+        for p in pinyin(
+            text, style=Style.TONE3, strict=False, neutral_tone_with_five=True
+        )
+    ]
+    for p in pinyins:
+        if p in lexicon:
+            phones += lexicon[p]
+        else:
+            phones.append("sp")
+
+    phones = "{" + " ".join(phones) + "}"
+    print("Raw Text Sequence: {}".format(text))
+    print("Phoneme Sequence: {}".format(phones))
+    sequence = np.array(text_to_sequence(phones, []))
+
+    return np.array(sequence)
+
+
+def synthesize(model, batchs):
+    print("batchs:", len(batchs))
+    for i, batch in enumerate(batchs):
+        output = model(p_control=1.0, e_control=1.0, d_control=1.0, **batch)
+        mel_prediction = output["mel_predictions"].asnumpy()
+        np.save("msfs2_%s.npy" % i, mel_prediction)
+
+
+def main():
+    args = parse_args()
+    hps = mindaudio.load_config(args.config)
```

Comment:
mindaudio load config function has been removed. please upload this scripts to examples/config.py

from pprint import pformat
import numpy as np
import yaml

class Config:
    def __init__(self, cfg_dict):
        for k, v in cfg_dict.items():
            if isinstance(v, (list, tuple)):
                setattr(self, k, [Config(x) if isinstance(x, dict) else x for x in v])
            else:
                setattr(self, k, Config(v) if isinstance(v, dict) else v)

    def __str__(self):
        return pformat(self.__dict__)

    def __repr__(self):
        return self.__str__()


def load_config(path):
    with open(path) as stream:
        hps = yaml.load(stream, yaml.Loader)
    hps = Config(hps)
    return hps